internal/system/agent/config_executor.go

```diff
@@ -0,0 +1,75 @@
+package agent
```

Comment:
Consider having a separate executor for get and set functionality.  

This would enable separate interfaces for each -- the contract currently requires a Context that's not used in all cases.  It would also allow you to eliminate the action parameter and integrate the get or set code directly into the same module.